core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -1208,24 +1250,30 @@ private void reloadSlow() {
         final long prevStructVersion = this.txFile.getStructureVersion();
         final long prevPartitionVersion = this.txFile.getPartitionTableVersion();
 
+        long deadline = this.configuration.getMicrosecondClock().getTicks() + this.configuration.getSpinLockTimeoutUs();
         // reload tx file, this will update the versions
-        this.readTxnSlow();
-        reloadStruct(prevStructVersion);
+        do {
+            this.readTxnSlow(deadline);
+        } while (!reloadStruct(prevStructVersion, txFile.getStructureVersion(), deadline));
```

Comment:
same, could be single loop

Reply:
Cannot be single loop, both read txn and read meta need independent retries